app/Http/Controllers/AdvertApplicationController.php

```diff
@@ -8,22 +8,35 @@
 use App\Models\Application;
 use App\Models\ApplicationAttachment;
 use App\Models\Asset;
+use App\Models\Company;
 use App\Models\User;
 use Illuminate\Http\Response;
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Gate;
 use Illuminate\Support\Facades\Log;
 
 class AdvertApplicationController extends Controller
 {
     /**
      * Display a listing of the resource.
      *
+     * @param int $company_id
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index($company_id)
```

Comment:
use a type hint:
```php
public function index(Company $company)
```

You won't need to make a query for the company then.

Reply:
using a company parameter seems not working